CHANGELOG.md

```diff
@@ -1,5 +1,282 @@
+ ```
+
+#### Initialization and finalization methods for a test ([#1108](https://github.com/DevExpress/testcafe/issues/1108))
+
+We have added the `before` and `after` methods to the test declaration.
+Use them to provide code that will be executed before a test is started and after it is finished.
+
+```js
+test
+    .before( async t => {
+        /* test initialization code */
+    })
+    ('My Test', async t => {
+        /* test code */
+    })
+    .after( async t => {
+        /* test finalization code */
+    });
+```
+
+#### Sharing variables between hooks and test code ([#841](https://github.com/DevExpress/testcafe/issues/841))
+
+You can now share variables between `fixture.beforeEach`, `fixture.afterEach`, `test.before`, `test.after` functions and test code
+by using the *test context* object.
+
+Test context is available through the `t.ctx` property.
+
+Instead of using a global variable, assign the object you want to share directly to `t.ctx` or create a property like in the following example.
+
+```js
+fixture `Fixture1`
+    .beforeEach(async t  => {
+          t.ctx.someProp = 123;
+     })
+    .afterEach(async t  => {
+          console.log(t.ctx.someProp); // > 123
+    });
+
+test
+    .before(async t => {
+         console.log(t.ctx.someProp); // > 123
+    })
+    ('Test1', async t => {
+        console.log(t.ctx.someProp); // > 123
+    });
+```
+
+#### Assertion methods to check for regexp match ([#1038](https://github.com/DevExpress/testcafe/issues/1038))
+
+We have added `match` and `notMatch` methods to check if a string matches a particular regular expression.
+
+```js
+await t.expect('foobar').match(/^f/, 'this assertion passes');
+```
+
+```js
+await t.expect('foobar').notMatch(/^b/, 'this assertion passes');
+```
+
+#### Improved filtering by predicates in selectors ([#1025](https://github.com/DevExpress/testcafe/issues/1025) and [#1065](https://github.com/DevExpress/testcafe/issues/1065))
+
+Selector's filter predicates now receive more information about the current node, which enables you to implement more advanced filtering logic.
+
+The `filter`, `find`, `parent`, `child` and `sibling` methods now pass the node's index to the predicate.
+The `find`, `parent`, `child` and `sibling` methods now also pass a node from the preceding selector.
+
+```js
+Selector('ul').find((node, idx, originNode) => {
+    // node === the <ul>'s descendant node
+    // idx === index of the current <ul>'s descendant node
+    // originNode === the <ul> element
+});
+```
+
+In addition, all these methods now allow you to pass objects to the predicate's scope on the client. To this end, we have added
+an optional `dependencies` parameter.
+
+```js
+const isNodeOk = ClientFunction(node => { /*...*/ });
+const aFlag = getFlag();
+
+Selector('ul').child(node => {
+    return isNodeOk(node) && aFlag;
+}, { isNodeOk, aFlag });
+```
+
+#### Filtering by negative index in selectors ([#738](https://github.com/DevExpress/testcafe/issues/738))
```

Comment:
May be better to specify list of methods which take negative index (e.g. `parent` method works another way)